File: src/main/java/com/extension/factcheck/api/AuthController.java
package com.extension.factcheck.api;

import com.extension.factcheck.application.AuthService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController
@RequestMapping("/auth")
public class AuthController {

  private final AuthService authService;

  @PostMapping
  public ResponseEntity<String> login(@RequestParam String googleId) {
    return ResponseEntity.ok().body(authService.login(googleId));
  }

}

File: src/main/kotlin/com/extension/factcheck/api/AuthController.kt
package com.extension.factcheck.api

import com.extension.factcheck.api.dto.IdTokenRequest
import com.extension.factcheck.application.AuthService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/v1/auth")
class AuthController(
    private val authService: AuthService,
) {

    @PostMapping
    fun login(@RequestBody request: IdTokenRequest): ResponseEntity<String> {
        return ResponseEntity.ok().body(authService.login(request.idToken))
    }
}
